ironic/drivers/modules/drac/inspect.py

```diff
@@ -35,16 +42,168 @@
+                           'Error details: %(err)'),
+                         {'code': job_response.status_code,
+                          'err': job_response.__dict__})
+                LOG.error(error)
+                raise exception.HardwareInspectionFailure(error=error)
+
+            job_id = job_response.__dict__['headers']['Location']
+            try:
+                job_id = re.search('JID_.+', job_id).group()
+            except drac_exceptions.BaseClientException as exc:
+                LOG.error('Unable to find the job id in response: %(err)',
+                          {'err': job_response.__dict__['_content']})
+                raise exception.HardwareInspectionFailure(error=exc)
+
+            start_time = datetime.now()
+
+            while True:
+                current_time = (datetime.now() - start_time)
+                xml_data = requests.get('%s%s%s' % (
+                    redfish_ip, _XML_DATA_URL, job_id), headers=_HEADERS,
+                    auth=(redfish_username, redfish_password), verify=False)
+
+                if _SYSTEM_CONFIG_TAG in str(xml_data.__dict__):
+                    xml_string = xml_data.__dict__.get('_content')
+                    root = ET.fromstring(xml_string)
+                    for child in root:
+                        nic_id = [child.attrib.get('FQDD') for ch in child
+                                  if ch.text == 'PXE']
+                        if nic_id:
+                            # Get MAC address of the given nic_id
+                            mac_address = self._get_mac_address(
+                                nic_id, redfish_ip,
+                                redfish_username, redfish_password)
+                            pxe_dev_nics.append(mac_address)
+                    return pxe_dev_nics
+                elif current_time.seconds >= _TIMEOUT_SECONDS:
+                    error = (_('The job to export the system configuration '
+                               'did not complete within %(second) seconds'),
+                             {'second': _TIMEOUT_SECONDS})
+                    raise exception.HardwareInspectionFailure(error=error)
+        return pxe_dev_nics
+
+    def _get_mac_address(self, nic_id, redfish_ip, redfish_username,
+                         redfish_password):
+        """Get MAC address of the given nic id.
+
+        :param nic_id: id of the network interface controller (NIC).
+        :param redfish_ip: IP address of the redfish driver.
+        :param redfish_username: username of the redfish driver.
+        :param redfish_password: password of the redfish driver.
+        :returns: Returns MAC address of the given nic id.
+        """
+        mac_response = requests.get('%s%s%s' % (
+            redfish_ip, _MAC_URL, nic_id[0]), auth=(
+            redfish_username, redfish_password), verify=False)
+        mac_address = json.loads(mac_response.__dict__.get(
+            '_content')).get('MACAddress')
+        return mac_address
```

Comment:
Let's return mac_address.upper() here so if some versions of the firmware happen to return the MAC in lower case, it won't matter.  See line 99 above...

Reply:
Done